File: examples/spring-boot-maven-plugin/src/main/java/community/flock/wirespec/examples/app/user/UserAdapter.java
package community.flock.wirespec.examples.app.user;

import java.util.List;

public interface UserAdapter {
    interface Module {
        UserAdapter getUserAdapter();
    }

    List<User> getAllUsers();

    User getUserByName(String name);

    User saveUser(User user);

    User deleteUserByName(String name);
}

File: examples/gradle-ktor/src/main/kotlin/community/flock/wirespec/example/gradle/app/user/UserAdapter.kt
package community.flock.wirespec.example.gradle.app.user

interface HasUserAdapter {
    val userAdapter: UserAdapter
}

interface UserAdapter {
    fun getAllUsers(): List<User>

    fun getUserByName(name: String): User

    fun saveUser(user: User): User

    fun deleteUserByName(name: String): User
}
